[Chow En Rong] iP

chownces opened the pull request:
## DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is __FREE__!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

chownces committed:
Complete level 1. Added Greet, Echo and Exit.

chownces committed:
Complete level 2. Added Add and List

chownces committed:
Complete level 3. Added mark as done feature.

chownces committed:
Refactor Duke.java to be more OOP-like, and use regex for matching input commands

chownces committed:
Complete level 4. Added Todos, Events and Deadlines

chownces committed:
Complete A-TextUiTesting. Added execute permission to runtest.sh, added text ui tests.

chownces committed:
Complete level 5. Added error handling

chownces committed:
Add javadoc comments

chownces committed:
Complete level 6. Added deletion functionality

chownces committed:
Add save to hard disk functionality

chownces committed:
Add LocalTime support for deadline dates

chownces committed:
Merge branch 'branch-Level-7'

chownces committed:
Merge branch 'branch-Level-8'

chownces committed:
Refactor program to utilize more OOP

chownces committed:
Organize classes into packages

chownces committed:
Add JUnit tests for Parser and TaskList classes

chownces committed:
Complete A-Jar

chownces committed:
Add JavaDoc comments

chownces committed:
Format code to align with coding standards

chownces committed:
Merge branch 'branch-A-JavaDoc'

chownces committed:
Merge branch 'branch-A-CodingStandard'

chownces committed:
Add find functionality

chownces committed:
Merge branch 'branch-Level-9'

chownces committed:
Fix coding style violations

chownces committed:
Merge branch 'add-gradle-support'

chownces committed:
Add checkstyle config and fix checkstyle warnings

chownces committed:
Modify build.gradle configuration to create JAR file correctly

chownces committed:
Add JavaFX GUI

chownces committed:
Merge branch 'branch-Level-10'

XXJJXJ commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,61 @@
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String LINE = "-----------------------------------------------------------------------\n";
+    private final Scanner sc;
```
Perhaps a more obvious name for the scanner would be better? Though I agree this should be intuitive enough.

XXJJXJ commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,83 @@
+            throw new InvalidFormatException("Please do not include commas!");
+        }
+
+        Pattern r = Pattern.compile(regex);
+        Matcher m = r.matcher(str);
```
Perhaps a more descriptive name would be good

weiquu replied:
Agree with XXJJXJ that you could consider more descriptive names. Perhaps something that includes regex and match in their names? That being said, I liked the use of regex here!

XXJJXJ reviewed (commented):
Wow! I can barely spot any issues with your code 👍 . If you think my comments are a bit forceful, feel free to ignore them.

ChuangZheQuan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+     * @param by          The duedate of this task.
+     * @param isDone      Whether this task is marked as done.
+     */
+    public Deadline(String description, LocalDate by, boolean isDone) {
```
Maybe use something more descriptive other than ```by``` ?

ChuangZheQuan commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    public Event(String description, String at) {
+        super(description);
+        this.at = at;
```
Maybe something descriptive other than ```by```?

ChuangZheQuan commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    public Event(String description, String at, boolean isDone) {
+        super(description, isDone);
+        this.at = at;
```
Maybe something descriptive other than ```at```?

ChuangZheQuan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,76 @@
+            return false;
+        }
+
+        // obj is an instance of Task, so it is okay to typecast here
```
Nice!

ChuangZheQuan reviewed (commented):
Really can't find any errors. Had to be very nitpick-y about it. Good job!

weiquu commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+ * This class encapsulates a deadline task in the Duke Application.
+ */
+public class Deadline extends Task {
+    private final LocalDate by;
```
I agree with Zhe Quan, this could possibly be something more descriptive. In addition, since this is a final variable that shouldn't be changed, you could consider capitalising the name to signal that. Perhaps something like `DUE_DATE`?

weiquu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class Duke {
+    private final Ui ui;
+    private final Storage storage;
```
It seems like you're declaring these as constants, in which case it might be better to capitalise the variable names? So something like `UI` and `STORAGE`.

weiquu commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,55 @@
+/**
+ * Thrown when reading a file with invalid format.
+ */
+class InvalidFileFormatException extends DukeException {
```
I liked the descriptiveness of the class names here!

weiquu commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,41 @@
+ * This class encapsulates an event task in the Duke Application.
+ */
+public class Event extends Task {
+    private final String at;
```
Similar to the comment for Deadline above! Perhaps could consider a more descriptive name as well as capitalising the variable since it's a constant?

weiquu commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,60 @@
+ * This class encapsulates the logic for saving and writing to the hard disk.
+ */
+public class Storage {
+    private final String filePath;
```
Same comment about final variables! Perhaps consider capitalising the name of the variable, like `FILEPATH`? That way it will signal that it is meant to be a constant.

weiquu commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,60 @@
+
+            FileWriter fw = new FileWriter(this.filePath);
+            fw.write(text);
+            fw.close();
```
Perhaps could consider a more descriptive name for the FileWriter?

weiquu commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,244 @@
+                Task newTask;
+                String[] values = line.split(",", -1);
+
+                // Checks to ensure correct line format
```
I liked the comments and clean code through the use of the switch statement!

weiquu commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,244 @@
+     * @throws InvalidTaskIndexException When the index is out of bounds.
+     * @throws InvalidFormatException    When the input command is of the wrong format.
+     */
+    public String done(String command) throws InvalidTaskIndexException, InvalidFormatException {
```
Perhaps a name like `markAsDone` would be better here? To align with the coding conventions of names presenting methods being verbs.

weiquu commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,244 @@
+     * @throws InvalidFormatException        When the input command is of the wrong format.
+     * @throws EmptyTodoDescriptionException When the description is empty.
+     */
+    public String todo(String command) throws InvalidFormatException, EmptyTodoDescriptionException {
```
Perhaps a name that is a verb would be better here? To align with the coding conventions of names presenting methods being verbs.

weiquu commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,244 @@
+     * @return Formatted string containing the added deadline task, to be displayed to the user.
+     * @throws InvalidFormatException When the input command is of the wrong format.
+     */
+    public String deadline(String command) throws InvalidFormatException {
```
Perhaps a name that is a verb would be better here? To align with the coding conventions of names presenting methods being verbs.

weiquu commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,244 @@
+     * @return Formatted string containing the added event task, to be displayed to the user.
+     * @throws InvalidFormatException When the input command is of the wrong format.
+     */
+    public String event(String command) throws InvalidFormatException {
```
Perhaps a name that is a verb would be better here? To align with the coding conventions of names presenting methods being verbs.

weiquu commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,148 @@
+
+import org.junit.jupiter.api.Test;
+
+public class ParserTest {
```
I liked the three part format used in naming the tests!

weiquu reviewed (commented):
Overall, I think that the code was really good, and I liked some of the methods used to validate commands as well as the OOP implementations (e.g. of errors). There were a couple of minor coding standard violations such as constant variables not being capitalised and some of the methods not being verbs, but in general coding standards were followed well.

chownces committed:
Add java assert statements

Java assert statements help to document important assumptions
that should hold at various points in the code. This leads to
less error prone code.

Let's add assertions to the classes that handle the Duke tasklist
logic.

The assertions are to be done in these classes as that is where the
main application logic resides.

chownces committed:
Merge pull request #1 from chownces/branch-A-Assertions

Add java assert statements

chownces committed:
Improve code quality

Improving code quality ensures better readability and maintainability
of the code base.

Let's refactor the code base to align it with the module code quality
guidelines.

chownces committed:
Merge pull request #2 from chownces/branch-A-CodeQuality

Improve code quality

chownces committed:
Add Github CI

chownces committed:
Implement archive feature

chownces committed:
Repackage duke files

chownces committed:
Refactor application commands into a new Command class

chownces committed:
Improve javafx GUI

chownces committed:
Add welcome text

chownces committed:
Add application title in JavaFx

chownces committed:
Add Ui.png

chownces committed:
Add user guide

chownces committed:
Set theme jekyll-theme-cayman

chownces committed:
Fix GFMD table formatting error

chownces committed:
Add data saving and editing data file sections to project README

chownces committed:
Update method names to better comply with coding standards

chownces committed:
Update instance fields to better comply with coding standards
